### ISSUE TITLE
KVar receiving incorrect initial value in KVal.map()

### ISSUE THREAD

**Issue by ethanmdavidson** [kwebio/kweb-core#65]
given this code:

```kotlin
fun main(args: Array<String>) {
    val count = KVar(0)

    Kweb(port = 8080) {
        doc.body.new {
            h1().text(count.map{ c -> "Count: $c"})
            button().text("Click me!").on.click {
                count.value += 1
            }
        }
    }
}
```

1) Open the page and click the button a few times (e.g. 4 times)
2) Reload the page
Expected behavior: page displays the proper count (e.g. 4)
Actual behavior: page displays "Count: 0"

The issue is due to the way KVar inherits from KVal. When you create the KVar, the super KVal is created with pValue = 0, and pValue doesn't get updated when the KVar is changed. Then, when you call map(), it returns a new KVal with a listener attached and an initial value of pValue. This can be further demonstrated by initializing the KVar with a different value, e.g. `val count = KVar(42)`. Every time the page is reloaded, it will initially render with 42.

**Comment by sanity** [comment]
Still getting my head around this, but is the issue that `KVar.value` change hook isn't updating `pValue` when `value` is modified - or does it go further than that?

**Comment by sanity** [comment]
Fix committed + unit test, and released in 0.4.21 - thanks guys!
